plugin/session_buffer.py

```diff
@@ -309,19 +311,24 @@ def purge_changes_async(self, view: sublime.View) -> None:
         sublime.set_timeout_async(lambda: self._on_after_change_async(view, version))
 
     def _on_after_change_async(self, view: sublime.View, version: int) -> None:
+        if self._is_saving:
+            self._has_changed_during_save = True
+            return
         self._do_color_boxes_async(view, version)
         self.do_semantic_tokens_async(view)
         if userprefs().link_highlight_style in ("underline", "none"):
             self._do_document_link_async(view, version)
         self.do_inlay_hints_async(view)
 
     def on_pre_save_async(self, view: sublime.View) -> None:
+        self._is_saving = True
```

Comment:
How do you guarantee this is always reset to False even in the face of unexpected errors?

Reply:
In case unexpected error happens during handling of `lsp_save_command`, the file will not be saved so I would consider that to be much bigger problem then the fact that we will prevent some optional requests from triggering.

As far as I can tell, there shouldn't be a case when save completes but something unexpected happens that prevents this from being called (or the "post*" one).

In the case the URI changes this won't be called but also the whole SessionBuffer will be recreated so it won't be an issue.